### PR TITLE
flac: add flac/1.3.4

### PR DESCRIPTION
**Pull request by Croydon**
Specify library name and version:  **flac/1.3.4**

**Note:** This uses now the official source code release. We can't do the same for 1.3.3 as the CMake scripts were not included in this one.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by prince-chrismc** [comment]
Super excited to see if https://github.com/conan-io/conan-center-index/pull/9707 kicks in 🤞

**Comment by uilianries** [comment]
There is a .pc file hidden in this package 🤨

**Comment by Croydon** [comment]
> ld: warning: ignoring file /usr/local/lib/libintl.dylib, building for macOS-arm64 but attempting to link with file built for macOS-x86_64
Undefined symbols for architecture arm64:
  "_libintl_gettext", referenced from:
      _share___getopt_internal in libgetopt.a(getopt.c.o)
ld: symbol(s) not found for architecture arm64

It looks like we don't have libintl in `libiconv` not `gettext` 🤔 
Not sure what to do here

File: recipes/flac/all/conanfile.py [hunk 1]
```diff
@@ -53,15 +55,14 @@ def source(self):
...
-        self._cmake.definitions["BUILD_DOCS"] = False
-        self._cmake.definitions["BUILD_TESTING"] = False
-        self._cmake.configure()
-        return self._cmake
+        cmake = CMake(self)
+        cmake.definitions["BUILD_EXAMPLES"] = False
+        cmake.definitions["BUILD_DOCS"] = False
+        cmake.definitions["BUILD_TESTING"] = False
+        cmake.configure()
+        return cmake
```
Comment (SpaceIm) [inline recipes/flac/all/conanfile.py:65]: so you can remove `_cmake = None`

**Comment by SpaceIm** [comment]
> > ld: warning: ignoring file /usr/local/lib/libintl.dylib, building for macOS-arm64 but attempting to link with file built for macOS-x86_64
> > Undefined symbols for architecture arm64:
> > "_libintl_gettext", referenced from:
> > _share___getopt_internal in libgetopt.a(getopt.c.o)
> > ld: symbol(s) not found for architecture arm64
> 
> It looks like we don't have libintl in `libiconv` nor `gettext` 🤔 And the pre-installed one is not there in the right architecture.. Not sure what to do here

libintl is libgettext

**Pull request closed without merging**